### PR TITLE
 Refactor the fileStore codes to fix the file creation failed and  derivative creation failed issue found during ingest. DAMSKEY-517

### PR DESCRIPTION
**Pull request by lsitu**
https://lib-jira.ucsd.edu:8443/browse/DAMSKEY-517

### CODE REVIEW

File: src/properties/log4j.properties [hunk 1]
```diff
@@ -4,3 +4,4 @@ log4j.appender.default.layout=org.apache.log4j.PatternLayout
 log4j.appender.default.layout.ConversionPattern=%d %p %c: %m%n
 log4j.logger.org.apache.solr.client=WARN
 log4j.logger.org.slf4j=ERROR
+log4j.logger.edu.ucsd.library.dams.file=DEBUG
```
Comment (escowles) [inline src/properties/log4j.properties]: Please revert this change -- we don't want debug logging to be merged.

**Comment by lsitu** [comment]
@escowles I've reverted the debug logging. Thanks.
